packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -693,25 +693,89 @@ contract StargateAdapter is
      */
     function lzCompose(
         address _from,
-        bytes32,
+        bytes32, // guid
         bytes calldata _message,
-        address,
-        bytes calldata
+        address, // caller
+        bytes calldata // extraData
     ) external payable override nonReentrant {
         // Verify caller is LayerZero endpoint
-        if (msg.sender != LZ_ENDPOINT) {
-            revert Unauthorized();
-        }
+        if (msg.sender != LZ_ENDPOINT) revert Unauthorized();
 
         // Validate the Stargate pool contract
         _validateStargatePool(_from);
 
-        // Extract the amount and compose message from OFT encoding
-        uint256 amount = OFTComposeMsgCodec.amountLD(_message);
-        bytes memory composeMsg = OFTComposeMsgCodec.composeMsg(_message);
+        // ---------------------------------------------------------------
+        // 1. Taxi header must be present
+        // ---------------------------------------------------------------
+        if (!TaxiCodec.isTaxi(_message)) revert InvalidMessage();
+        /// forge-lint: disable-start(mixed-case-variable)
+        // msg layout: <type><assetId><receiver><amountSD><bytes32 sender><composeMsg>
+        (, , uint64 amtSD, bytes memory taxiPayload) = TaxiCodec.decodeTaxi(
+            _message
+        );
+        /// forge-lint: disable-end(mixed-case-variable)
```

Comment:
let's get rid of that as we're not uisng the forge linter - we're using solhint across the monorepo